sdk/src/main/kotlin/io/rover/sdk/ui/blocks/poll/VotingStorage.kt

```diff
@@ -0,0 +1,49 @@
+package io.rover.sdk.ui.blocks.poll
+
+import io.rover.sdk.logging.log
+import io.rover.sdk.platform.KeyValueStorage
+import org.json.JSONException
+import org.json.JSONObject
+import java.lang.Exception
+
+internal class VotingStorage(private val keyValueStorage: KeyValueStorage) {
+    fun setPollResults(pollId: String, value: String) {
+        keyValueStorage["$pollId-results"] = value
+    }
+
+    fun incrementSavedPollState(pollId: String, optionId: String) {
+        keyValueStorage["$pollId-vote"] = optionId
+
+        val optionResultsJson = keyValueStorage["$pollId-results"]
+
+        optionResultsJson?.let {
+            try {
+                val optionResults = OptionResults.decodeJson(JSONObject(it))
+                val resultsMap = optionResults.results.toMutableMap()
+                resultsMap[optionId] = resultsMap[optionId]!!.plus(1)
+                val resultsToInsert = optionResults.copy(results = resultsMap)
+
+                keyValueStorage["$pollId-results"] = resultsToInsert.encodeJson().toString()
+            } catch (e: JSONException) {
+                log.w("Poll JSON decode problem details: $e, ${e.stackTrace.joinToString("\n")}")
```

Comment:
Hm, why is the full stack trace log needed here?

Reply:
I added it because I saw it was being used elsewhere in the catch statements for json exceptions with a comment explaining that it's useful for diagnosing JSON decode errors. Although not sure why it's added to general exception catch statement. I'll remove from all